plugins/nlp/nlp_eng.cpp

```diff
@@ -0,0 +1,75 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "nlp_eng.hpp"
+#include "nlp_engine.h"
+
+static NLP_ENGINE *nlpEngine = NULL;
+
+NLPEng::NLPEng(string workingFolderPath) {
+    if (!workingFolderPath.empty()) {
+        workingFolder = workingFolderPath;
+    } else {
+        workingFolder.clear();
+    }
+
+#ifdef NLP_DEBUG    
+    ofstream handle;
+    handle.open(NLP_DEBUG_FILE, std::ofstream::out | std::ofstream::app);
+    handle << "[workingFolder: " << workingFolder << "]" << endl;
+    handle.close();
+#endif
+}
+
+NLPEng::~NLPEng() {
+    if (nlpEngine != NULL) {
```

Comment:
This logic does not seem right.
- If you can have multiple instances of NLPEng, then after the first one is deleted the second one will crash because nlpEngine will be non-null
- If you cannot have multiple instances sharing an nlpEngine, then it would be better as a member variable, rather than a static.